modules/rgbd/src/hash_tsdf.cpp

```diff
@@ -127,15 +363,27 @@ void HashTSDFVolumeCPU::integrate(InputArray _depth, float depthFactor, const Ma
     parallel_for_(allocateRange, AllocateVolumeUnitsInvoker);
 
     //! Perform the allocation
-    int res = volumeUnitResolution;
-    Point3i volumeDims(res, res, res);
     for (auto idx : newIndices)
     {
         VolumeUnit& vu = volumeUnits[idx];
         Matx44f subvolumePose = pose.translate(volumeUnitIdxToVolume(idx)).matrix;
-        vu.pVolume = makePtr<TSDFVolumeCPU>(voxelSize, subvolumePose, raycastStepFactor, truncDist, maxWeight, volumeDims);
+        
+        vu.pose = subvolumePose;
+        vu.index = lastVolIndex; lastVolIndex++;
+        if (lastVolIndex > volUnitsData.size().height)
+        {
+            //std::cout << "    +"  << std::endl;
+            volUnitsData.resize(lastVolIndex - 1 + VOLUMES_SIZE);
```

Comment:
If increasing the `Mat` such a way, there will be reallocation at each resize.
To avoid that, add `reserve(currentSize * reserveFactor)` call before resizing.

Reply:
^^^Disregard that, in my case `reserve` would be called at each resize leading to constant reallocation too.
Instead, a new size should be quantized to a grid of values, i.e. rounded up to a power of `reserveFactor` (this is why it's easier to set `reserveFactor` to 2)